services/crates/crates-license.spec.js

```diff
@@ -0,0 +1,27 @@
+import { expect } from 'chai'
+import { InvalidResponse } from '../index.js'
+import CratesLicense from './crates-license.service.js'
+
+describe('CratesLicense', function () {
+  it('throws InvalidResponse on error response', function () {
+    expect(() =>
+      CratesLicense.transform({ errors: [{ detail: 'invalid semver' }] })
+    )
+      .to.throw(InvalidResponse)
+      .with.property('prettyMessage', 'invalid semver')
+  })
+
+  it('throws InvalidResponse on null license with specific version', function () {
+    expect(() =>
+      CratesLicense.transform({ version: { num: '1.2.3', license: null } })
+    )
+      .to.throw(InvalidResponse)
+      .with.property('prettyMessage', 'invalid null license')
+  })
+
+  it('throws InvalidResponse on null license with latest version', function () {
+    expect(() => CratesLicense.transform({ versions: [{ license: null }] }))
+      .to.throw(InvalidResponse)
+      .with.property('prettyMessage', 'invalid null license')
+  })
+})
```

Comment:
Should we have an explicit test for the valid case here, or have you left it because its covered by the service tests?

Reply:
To be honest I hadn't really thought about it. I started out with the unit tests, then decided to add the service tests to account for the schema, and did so without realizing the service tests make these unit tests largely duplicative.

Do you think it's worth even keeping these?

Reply:
I think yes - we're covering a wider range of cases here. Also its more explicit what we're really testing here. In fact...